### PR TITLE
Misc documentation cleanup

### PR DESCRIPTION
**Pull request by timar**
Remove obsolete info, some formatting improvements and reorganization. To be continued... So this is the first cut.

### CODE REVIEW

File: README.CONTRIBUTING.md [hunk 1]
```diff
@@ -64,3 +64,69 @@ When reviewing a patch, we look for the following information in the commit mess
...
+  explicitly.
+
+### JavaScript
+
+- Indent code with TABs.
+
+- For new code, use TypeScript.
+
+Security credential related changes
+-----------------------------------
```
Comment (vmiklos) [inline README.CONTRIBUTING.md:127]: Nit: not sure if this is valid markdown.

**Review by vmiklos — APPROVED** [review]
I don't see anything obviously problematic with these changes. See the inline comment for a nit.

File: README.CONTRIBUTING.md (hunk 1 above)
Comment (timar) [reply README.CONTRIBUTING.md:127]: Good point... it's different than others. But I did not see difference in preview.

Comment (timar) [reply README.CONTRIBUTING.md]: "fixed"

**Pull request merged**